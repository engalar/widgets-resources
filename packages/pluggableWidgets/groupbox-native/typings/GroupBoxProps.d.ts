/**
 * This file was generated from GroupBox.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, ReactNode } from "react";
import { DynamicValue, NativeIcon } from "mendix";

export type CollapsibleEnum = "collapsibleNo" | "collapsibleYesExpanded" | "collapsibleYesCollapsed";

export interface GroupBoxProps<Style> {
    name: string;
    style: Style[];
    headerContent?: ReactNode;
    content: ReactNode;
    showHeader: boolean;
    collapsible: CollapsibleEnum;
    iconCollapsed?: DynamicValue<NativeIcon>;
    iconExpanded?: DynamicValue<NativeIcon>;
}

export interface GroupBoxPreviewProps {
    class: string;
    style: string;
    headerContent: { widgetCount: number; renderer: ComponentType<{caption?: string}> };
    content: { widgetCount: number; renderer: ComponentType<{caption?: string}> };
    showHeader: boolean;
    collapsible: CollapsibleEnum;
    iconCollapsed: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null;
    iconExpanded: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null;
}
