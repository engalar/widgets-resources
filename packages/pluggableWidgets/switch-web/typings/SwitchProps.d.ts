/**
 * This file was generated from Switch.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { CSSProperties } from "react";
import { EditableValue } from "mendix";

export type DeviceStyleEnum = "auto" | "android" | "iOS";

export interface SwitchContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    id: string;
    booleanAttribute: EditableValue<boolean>;
    deviceStyle: DeviceStyleEnum;
}

export interface SwitchPreviewProps {
    class: string;
    style: string;
    booleanAttribute: string;
    deviceStyle: DeviceStyleEnum;
}
