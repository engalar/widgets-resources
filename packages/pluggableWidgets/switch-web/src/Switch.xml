<?xml version="1.0" encoding="utf-8"?>
<widget id="com.mendix.widget.custom.switch.Switch" needsEntityContext="true" offlineCapable="true" pluginWidget="true"
        xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../node_modules/mendix/custom_widget.xsd">
    <name>Switch</name>
    <description>Toggle a boolean attribute</description>
    <icon>
        iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAMAAACdt4HsAAAAY1BMVEUAAABV1f9VVVViYmJVVVVXV1dXV1dNsvlYWFhVVVVVVVVVVVVVVVVVVVVJsfhIsfhWVlZVVVVJsfhWVlZWVlZJsfhVVVVWVlZVVVVVVVVIsPhVVVVWVlZJsPhWVlZIsPdVVVUhgB8dAAAAH3RSTlMABgwNISMmKzE/VFp1gY+Uqq69vsfV2Nns8vPz+v39w1yAsQAAAP9JREFUeNrtlcsSgjAMRS/I+40IKIrw/19pgzMdKprByYZFzy5Mc2lumxQWi8VyUKK86e8zw71v8gi/SLp5F12Cb3iXeTcXDxvi6/wH13iT/6DvjyoNXDC4QVq9V34oeMv/ax878OtlD2YVVP8zA8KiHYa2CMGSPckHw3/SzOCU47Qwlg6vQOvXZ9HR/uGcJ82ZV6AqOmgicsVHOa0oeR/IyUiHuYoqhONaYOR9qFRKrqNGRSmKyaAAR6pSGh31KgrQmgItOAKV0uuI+sfFYAoM4HCpsyQCJ5VyE5cgNlF8jOKLJL3K4maStrN4oEhHmnSoSse68GGRPm0Wi8VyTF4UHle0saM+KAAAAABJRU5ErkJggg==
    </icon>
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="Data source">
                <property key="booleanAttribute" type="attribute">
                    <caption>Boolean attribute</caption>
                    <description>Attribute to toggle</description>
                    <attributeTypes>
                        <attributeType name="Boolean"/>
                    </attributeTypes>
                </property>
            </propertyGroup>
            <propertyGroup caption="Style">
                    <property key="deviceStyle" type="enumeration" defaultValue="auto">
                        <caption>Platform</caption>
                        <description>The general appearance of the switch. With auto detect, on the web it will be iOS
                        </description>
                        <enumerationValues>
                            <enumerationValue key="auto">Auto detect</enumerationValue>
                            <enumerationValue key="android">Android</enumerationValue>
                            <enumerationValue key="iOS">iOS</enumerationValue>
                        </enumerationValues>
                    </property>
                    <property key="colorStyle" type="enumeration" defaultValue="default">
                        <caption>Switch style</caption>
                        <description>The color style of the switch</description>
                        <enumerationValues>
                            <enumerationValue key="default">Widget default</enumerationValue>
                            <enumerationValue key="primary">Primary</enumerationValue>
                            <enumerationValue key="inverse">Inverse</enumerationValue>
                            <enumerationValue key="success">Success</enumerationValue>
                            <enumerationValue key="info">Info</enumerationValue>
                            <enumerationValue key="warning">Warning</enumerationValue>
                            <enumerationValue key="danger">Danger</enumerationValue>
                        </enumerationValues>
                    </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Common">
            <systemProperty key="Label"/>
            <systemProperty key="Editability"/>
    </propertyGroup>
    </properties>
</widget>
