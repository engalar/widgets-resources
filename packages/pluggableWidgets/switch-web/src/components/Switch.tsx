import { MouseEvent, KeyboardEvent, createElement, FunctionComponent } from "react";
import classNames from "classnames";
import { Alert } from "@mendix/piw-utils-internal";
import { SwitchContainerProps } from "../../typings/SwitchProps";

export interface SwitchProps extends Pick<SwitchContainerProps, "id" | "class" | "style" | "tabIndex" | "deviceStyle"> {
    onClick?: (event: MouseEvent<HTMLDivElement>) => void;
    onKeyDown?: (event: KeyboardEvent<HTMLDivElement>) => void;
    isChecked: boolean;
    editable: boolean;
    validation: string | undefined;
}

export const Switch: FunctionComponent<SwitchProps> = props => {
    return (
        <div className={classNames("widget-switch", props.class, props.deviceStyle)} style={props.style}>
            <input
                checked={props.isChecked}
                className={classNames("widget-switch-checkbox", { enabled: props.editable })}
                readOnly
                type="checkbox"
            />
            <div
                className={classNames("widget-switch-btn-wrapper", "widget-switch-btn-wrapper-default", {
                    checked: props.isChecked,
                    disabled: !props.editable,
                    "un-checked": !props.isChecked
                })}
                onClick={props.onClick}
                onKeyDown={props.onKeyDown}
                tabIndex={props.tabIndex ?? 0}
                role="checkbox"
                aria-checked={props.isChecked}
                aria-labelledby={props.id}
            >
                <small
                    className={classNames("widget-switch-btn", {
                        left: !props.isChecked,
                        right: props.isChecked
                    })}
                />
            </div>
            <Alert bootstrapStyle={"danger"}>{props.validation}</Alert>
        </div>
    );
};
