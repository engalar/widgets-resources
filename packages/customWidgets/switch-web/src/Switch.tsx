import { SFC, createElement, KeyboardEvent } from "react";
import classNames from "classnames";

import { Alert } from "./components/Alert";
import { ColorStyle, DeviceStyle } from "./components/SwitchContainer";

import "./ui/Switch.scss";

export interface SwitchProps {
    alertMessage?: string;
    colorStyle: ColorStyle;
    className?: string;
    deviceStyle?: DeviceStyle;
    isChecked: boolean;
    onClick: () => void;
    status: SwitchStatus;
    style?: object;
    labelId?: string;
}

export type SwitchStatus = "enabled" | "disabled" | "no-context";

export const Switch: SFC<SwitchProps> = props => (
    <div className={classNames("widget-switch", props.className, props.deviceStyle)} style={props.style}>
        <input
            checked={props.isChecked}
            className={classNames("widget-switch-checkbox", { enabled: props.status === "enabled" })}
            readOnly={true}
            type={"checkbox"}
        />
        ,
        <div
            className={classNames(`widget-switch-btn-wrapper widget-switch-btn-wrapper-${props.colorStyle}`, {
                checked: props.isChecked,
                disabled: props.status === "disabled",
                "no-switch": props.status === "no-context",
                "un-checked": !props.isChecked
            })}
            onClick={props.status === "enabled" ? props.onClick : undefined}
            onKeyDown={(e: KeyboardEvent<HTMLDivElement>) => {
                if (props.status === "enabled" && e.key === " ") {
                    e.preventDefault();
                    props.onClick();
                }
            }}
            tabIndex={0}
            role={"checkbox"}
            aria-checked={props.isChecked}
            aria-labelledby={props.labelId}
        >
            <small
                className={classNames("widget-switch-btn", {
                    left: !props.isChecked,
                    right: props.isChecked
                })}
            />
        </div>
        <Alert message={props.alertMessage} />
    </div>
);

Switch.defaultProps = {
    colorStyle: "default",
    deviceStyle: "auto"
};

Switch.displayName = "Switch";
