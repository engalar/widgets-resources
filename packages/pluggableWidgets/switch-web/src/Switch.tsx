import { createElement, KeyboardEvent, FunctionComponent } from "react";
import classNames from "classnames";

import { SwitchContainerProps } from "../typings/SwitchProps";
import { Alert } from "./components/Alert";

// import "./ui/Switch.scss";
import { isAvailable } from "@mendix/piw-utils-internal";

// note: it looks like "no-context" is not possible anymore as in framework an attribute is always available, but is unavailable, loading, or available.
// todo: should i move system prop visibility to separate group? or is it already there?
// todo: do we want to keep "default" style?

export type SwitchStatus = "enabled" | "disabled" | "no-context";

export const Switch: FunctionComponent<SwitchContainerProps> = props => {
    const isChecked = isAvailable(props.booleanAttribute);
    const editable = !props.booleanAttribute.readOnly;

    return (
        <div className={classNames("widget-switch", props.class, props.deviceStyle)} style={props.style}>
            <input
                checked={isChecked}
                className={classNames("widget-switch-checkbox", { enabled: editable })}
                readOnly={true}
                type={"checkbox"}
            />
            ,
            <div
                className={classNames(
                    `widget-switch-btn-wrapper`,
                    {
                        "widget-switch-btn-wrapper-default": !props.class?.match(
                            /widget-switch-btn-wrapper-{primary|secondary|success|warning|danger}/
                        )
                    },
                    props.class,
                    {
                        checked: isChecked,
                        disabled: !editable,
                        "un-checked": !isChecked
                    }
                )}
                // onClick={editable ? props.onClick : undefined} // todo: execute action
                // onKeyDown={(e: KeyboardEvent<HTMLDivElement>) => {
                //     if (editable && e.key === " ") {
                //         e.preventDefault();
                //         props.onClick(); // todo: execute action
                //     }
                // }}
                tabIndex={0}
                role={"checkbox"}
                aria-checked={isChecked}
                aria-labelledby={props.id}
            >
                <small
                    className={classNames("widget-switch-btn", {
                        left: !isChecked,
                        right: isChecked
                    })}
                />
            </div>
            <Alert message={props.booleanAttribute.validation} />
        </div>
    );
};

Switch.displayName = "Switch";
