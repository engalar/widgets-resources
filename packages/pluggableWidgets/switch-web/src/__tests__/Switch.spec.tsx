import { actionValue, EditableValueBuilder, Alert, AlertProps } from "@mendix/piw-utils-internal";
import { mount, ReactWrapper } from "enzyme";
import { createElement } from "react";

import SwitchComponent, { SwitchProps } from "../components/Switch";
import { SwitchContainerProps } from "../../typings/SwitchProps";
import Switch from "../Switch";

describe("Switch", () => {
    let switchWrapper: ReactWrapper<SwitchContainerProps, any>;
    let switchComponent: ReactWrapper<SwitchProps, any>;
    let switchComponentWrapper: ReactWrapper<any, any>;
    let checkbox: ReactWrapper<any, any>;
    let switchButtonWrapper: ReactWrapper<any, any>;
    let switchButton: ReactWrapper<any, any>;
    let alert: ReactWrapper<AlertProps, any>;
    const createAndFindElements = (props: SwitchContainerProps): void => {
        switchWrapper = mount(createElement(Switch, props));
        switchComponent = switchWrapper.find(SwitchComponent);
        switchComponentWrapper = switchComponent.find(".widget-switch");
        checkbox = switchComponent.find(".widget-switch-checkbox");
        switchButtonWrapper = switchComponent.find(".widget-switch-btn-wrapper");
        switchButton = switchComponent.find(".widget-switch-btn");
        alert = switchComponent.find(Alert);
    };
    const createProps = (props?: Partial<SwitchContainerProps>): SwitchContainerProps => {
        const defaultProps: SwitchContainerProps = {
            name: "switch",
            class: "company-switch-widget",
            style: { color: "red" },
            tabIndex: 0,
            id: "com.mendix.widgets.custom.switch1",
            booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build(),
            action: undefined,
            deviceStyle: "auto"
        };

        return { ...defaultProps, ...props };
    };

    it("with editable value should render the structure correctly", () => {
        createAndFindElements(
            createProps({
                booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
            })
        );

        expect(switchWrapper).toMatchSnapshot();
    });

    it("with readonly value should render the structure correctly", () => {
        createAndFindElements(
            createProps({
                booleanAttribute: new EditableValueBuilder<boolean>().isReadOnly().withValue(false).build()
            })
        );

        expect(switchWrapper).toMatchSnapshot();
    });

    it("without validation message should render correctly", () => {
        createAndFindElements(
            createProps({
                booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
            })
        );

        expect(alert).toMatchSnapshot();
    });

    it("with validation message should render correctly", () => {
        createAndFindElements(
            createProps({
                booleanAttribute: new EditableValueBuilder<boolean>().withValidation("error").withValue(false).build()
            })
        );

        expect(alert).toMatchSnapshot();
    });

    it("when value is false should have the correct checkbox attribute", () => {
        createAndFindElements(createProps());

        expect(checkbox.props().checked).toBe(false);
    });

    it("when value is true should have the correct checkbox attribute", () => {
        createAndFindElements(
            createProps({ booleanAttribute: new EditableValueBuilder<boolean>().withValue(true).build() })
        );

        expect(checkbox.props().checked).toBe(true);
    });

    it("with iOS device style should render correct class", () => {
        createAndFindElements(createProps({ deviceStyle: "iOS" }));
        console.log(switchComponentWrapper);
        expect(switchComponentWrapper.hasClass("iOS")).toBe(true);
    });

    it("with android device style should render correct class", () => {
        createAndFindElements(createProps({ deviceStyle: "android" }));

        expect(switchComponentWrapper.hasClass("android")).toBe(true);
    });

    it("with auto device style should render correct class", () => {
        createAndFindElements(createProps({ deviceStyle: "auto" }));

        expect(switchComponentWrapper.hasClass("auto")).toBe(true);
    });

    it("with tabIndex passed renders correctly", () => {
        createAndFindElements(createProps({ tabIndex: 1 }));

        expect(switchButtonWrapper.props().tabIndex).toEqual(1);
    });

    it("without tabIndex passed renders correctly", () => {
        createAndFindElements(createProps({ tabIndex: undefined }));

        expect(switchButtonWrapper.props().tabIndex).toEqual(0);
    });

    describe("when editable", () => {
        it("should render elements with correct attributes", () => {
            createAndFindElements(createProps());

            expect(checkbox.hasClass("enabled")).toBe(true);
            expect(switchButtonWrapper.hasClass("enabled")).toBe(false);
        });

        it("should invoke preventDefault onClick", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);
            const eventMock = { preventDefault: jest.fn() };

            switchButtonWrapper.simulate("click", eventMock);

            expect(eventMock.preventDefault).toHaveBeenCalled();
        });

        it("should invoke preventDefault on space keydown", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);
            const eventMock = { preventDefault: jest.fn(), key: " " };

            switchButtonWrapper.simulate("keydown", eventMock);

            expect(eventMock.preventDefault).toHaveBeenCalled();
        });

        it("should invoke action on click", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);

            switchButtonWrapper.simulate("click");

            expect(props.action?.execute).toHaveBeenCalled();
        });

        it("should invoke action on space keydown", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);

            switchButtonWrapper.simulate("keydown", { key: " " });

            expect(props.action?.execute).toHaveBeenCalled();
        });

        it("should not invoke action on keydown of any key but space", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);

            switchButtonWrapper.simulate("keydown", { key: "enter" });

            expect(props.action?.execute).not.toHaveBeenCalled();
        });

        describe("when value is available", () => {
            it("should toggle the attributes value onClick", () => {
                const props = createProps({
                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
                });
                createAndFindElements(props);

                switchButton.simulate("click");

                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
                expect(props.booleanAttribute.value).toEqual(true);

                switchButton.simulate("click");

                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
                expect(props.booleanAttribute.value).toEqual(false);
            });

            it("should toggle the attribute value on space keydown", () => {
                const props = createProps({
                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
                });
                createAndFindElements(props);

                switchButton.simulate("keydown", { key: " " });

                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
                expect(props.booleanAttribute.value).toEqual(true);

                switchButton.simulate("keydown", { key: " " });

                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
                expect(props.booleanAttribute.value).toEqual(false);
            });

            it("should not toggle the attribute on keydown of any key but space", () => {
                const props = createProps({
                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
                });
                createAndFindElements(props);

                switchButton.simulate("keydown", { key: "enter" });

                expect(props.booleanAttribute.setValue).not.toHaveBeenCalled();
            });
        });
    });

    describe("when readonly", () => {
        it("should render elements with correct attributes", () => {
            createAndFindElements(
                createProps({ booleanAttribute: new EditableValueBuilder<boolean>().isReadOnly().build() })
            );

            expect(checkbox.hasClass("enabled")).toBe(false);
            expect(switchButtonWrapper.hasClass("disabled")).toBe(true);
        });

        it("should not invoke action", () => {
            const props = createProps({
                booleanAttribute: new EditableValueBuilder<boolean>().isReadOnly().build(),
                action: actionValue()
            });
            createAndFindElements(props);

            switchButton.simulate("click");

            expect(props.action?.execute).not.toHaveBeenCalled();
        });

        it("should not change the attributes value", () => {
            const props = createProps({
                booleanAttribute: new EditableValueBuilder<boolean>().isReadOnly().build()
            });
            createAndFindElements(props);

            switchButton.simulate("click");

            expect(props.booleanAttribute.setValue).not.toHaveBeenCalled();
        });

        it("should invoke preventDefault onClick", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);
            const eventMock = { preventDefault: jest.fn() };

            switchButtonWrapper.simulate("click", eventMock);

            expect(eventMock.preventDefault).toHaveBeenCalled();
        });

        it("should invoke preventDefault on keydown space", () => {
            const props = createProps({ action: actionValue() });
            createAndFindElements(props);
            const eventMock = { preventDefault: jest.fn(), key: " " };

            switchButtonWrapper.simulate("keydown", eventMock);

            expect(eventMock.preventDefault).toHaveBeenCalled();
        });
    });
});
