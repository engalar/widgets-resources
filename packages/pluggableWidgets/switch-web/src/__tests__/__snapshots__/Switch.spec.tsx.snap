// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch with editable value should render the structure correctly 1`] = `
<Switch
  booleanAttribute={
    Object {
      "displayValue": "Formatted false",
      "formatter": Object {
        "config": Object {},
        "format": [MockFunction] {
          "calls": Array [
            Array [
              false,
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "Formatted false",
            },
          ],
        },
        "getFormatPlaceholder": [MockFunction],
        "parse": [MockFunction],
        "type": "datetime",
        "withConfig": [MockFunction],
      },
      "readOnly": false,
      "setFormatter": [MockFunction],
      "setTextValue": [MockFunction],
      "setValidator": [MockFunction],
      "setValue": [MockFunction],
      "status": "available",
      "validation": undefined,
      "value": false,
    }
  }
  class="company-switch-widget"
  deviceStyle="auto"
  id="com.mendix.widgets.custom.switch1"
  name="switch"
  style={
    Object {
      "color": "red",
    }
  }
  tabIndex={0}
>
  <Switch
    class="company-switch-widget"
    deviceStyle="auto"
    editable={true}
    id="com.mendix.widgets.custom.switch1"
    isChecked={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
    tabIndex={0}
  >
    <div
      className="widget-switch company-switch-widget auto"
      style={
        Object {
          "color": "red",
        }
      }
    >
      <input
        checked={false}
        className="widget-switch-checkbox enabled"
        readOnly={true}
        type="checkbox"
      />
      <div
        aria-checked={false}
        aria-labelledby="com.mendix.widgets.custom.switch1"
        className="widget-switch-btn-wrapper widget-switch-btn-wrapper-default un-checked"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="checkbox"
        tabIndex={0}
      >
        <small
          className="widget-switch-btn left"
        />
      </div>
      <Alert
        bootstrapStyle="danger"
      />
    </div>
  </Switch>
</Switch>
`;

exports[`Switch with readonly value should render the structure correctly 1`] = `
<Switch
  booleanAttribute={
    Object {
      "displayValue": "Formatted false",
      "formatter": Object {
        "config": Object {},
        "format": [MockFunction] {
          "calls": Array [
            Array [
              false,
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "Formatted false",
            },
          ],
        },
        "getFormatPlaceholder": [MockFunction],
        "parse": [MockFunction],
        "type": "datetime",
        "withConfig": [MockFunction],
      },
      "readOnly": true,
      "setFormatter": [MockFunction],
      "setTextValue": [MockFunction],
      "setValidator": [MockFunction],
      "setValue": [MockFunction],
      "status": "available",
      "validation": undefined,
      "value": false,
    }
  }
  class="company-switch-widget"
  deviceStyle="auto"
  id="com.mendix.widgets.custom.switch1"
  name="switch"
  style={
    Object {
      "color": "red",
    }
  }
  tabIndex={0}
>
  <Switch
    class="company-switch-widget"
    deviceStyle="auto"
    editable={false}
    id="com.mendix.widgets.custom.switch1"
    isChecked={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
    tabIndex={0}
  >
    <div
      className="widget-switch company-switch-widget auto"
      style={
        Object {
          "color": "red",
        }
      }
    >
      <input
        checked={false}
        className="widget-switch-checkbox"
        readOnly={true}
        type="checkbox"
      />
      <div
        aria-checked={false}
        aria-labelledby="com.mendix.widgets.custom.switch1"
        className="widget-switch-btn-wrapper widget-switch-btn-wrapper-default disabled un-checked"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="checkbox"
        tabIndex={0}
      >
        <small
          className="widget-switch-btn left"
        />
      </div>
      <Alert
        bootstrapStyle="danger"
      />
    </div>
  </Switch>
</Switch>
`;

exports[`Switch with validation message should render correctly 1`] = `
<Alert
  bootstrapStyle="danger"
>
  <div
    className="alert alert-danger"
  >
    error
  </div>
</Alert>
`;

exports[`Switch without validation message should render correctly 1`] = `
<Alert
  bootstrapStyle="danger"
/>
`;
